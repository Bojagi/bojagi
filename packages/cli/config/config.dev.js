module.exports = {
  uploadApiUrl: 'https://upload.dev.bojagi.io',
};
