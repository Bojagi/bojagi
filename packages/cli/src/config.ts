import * as fs from 'fs';
import { defaultConfig } from './defaultConfig';
import getCiSettingsFactory, { CiSettings } from './utils/getCiSettings';

import simpleGit = require('simple-git/promise');

const { PWD } = process.env;
const CONFIG_FILE_PRIO = [
  {
    path: `${PWD}/.bojagirc.js`,
    fn: loadJsConfig,
  },
  {
    path: `${PWD}/.bojagirc.json`,
    fn: loadJsonConfig,
  },
  {
    path: `${PWD}/.bojagirc`,
    fn: loadJsonConfig,
  },
];

export type CollectorTuple = [string, Record<string, any>];

export type BaseConfig = {
  componentMarker: string;
  dir: string;
  dryRun: boolean;
  webpackConfig: string;
  executionPath: string;
  decoratorPath: string;
  storyPath: string;
  uploadApiUrl: string;
  previewPort: number;
  previewDownloadUrl: string;
  collectors: (string | CollectorTuple)[];
};

export type Config = CiSettings & BaseConfig;

const getCiSettings = getCiSettingsFactory(process.env);

export const getConfig: () => Promise<Config> = async () => {
  const configFile = loadConfigFile();
  return {
    ...(await getGitSettings(configFile.executionPath || defaultConfig.executionPath)),
    ...defaultConfig,
    ...configFile,
    ...getCiSettings(),
  };
};

async function getGitSettings(executionPath: string) {
  try {
    const git = simpleGit(executionPath);
    const result = await git.log();
    return {
      commit: result.latest.hash,
    };
  } catch {
    return {};
  }
}

function loadConfigFile() {
  const foundConfigFile = CONFIG_FILE_PRIO.find(fc => fs.existsSync(fc.path));
  if (foundConfigFile) {
    return foundConfigFile.fn(foundConfigFile.path);
  }
  return {};
}

function loadJsConfig(path) {
  return require(path);
}

function loadJsonConfig(path) {
  return JSON.parse(fs.readFileSync(path).toString());
}
