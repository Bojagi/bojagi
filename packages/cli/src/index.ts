import setEnv from './setEnv';
import bundle from './commands/bundle';
import deploy from './commands/deploy';
import upload from './commands/upload';
import runCollectors from './commands/runCollectors';
import list from './commands/list';
import preview from './commands/preview';
import scan from './commands/scan';
import cleanup from './commands/cleanup';
import init from './commands/init';
import docs from './commands/docs';

import program = require('commander');

setEnv();

bundle(program);
deploy(program);
upload(program);
runCollectors(program);
list(program);
preview(program);
scan(program);
cleanup(program);
init(program);
docs(program);

program.parse(process.argv);
