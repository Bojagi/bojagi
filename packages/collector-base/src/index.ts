import writeRegisteredPropsFactory from './writeRegisteredProps';
import { propsRegistry } from './propsRegistry';

export * from './createProp';
export * from './propsRegistry';
export * from './createFilePropList';
export * from './writeProps';
export * from './getComponents';
export * from './componentsAvailable';
export * from './types';

export const writeRegisteredProps = writeRegisteredPropsFactory({
  propsRegistry,
});
