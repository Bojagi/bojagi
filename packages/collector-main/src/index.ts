import {
  getComponents,
  addProps,
  writeRegisteredProps,
  CollectorMetadata,
  CollectorFunctionOptions,
} from '@bojagi/collector-base';
import * as fs from 'fs';
import { promisify } from 'util';
import { getWebpackConfig } from './getWebpackConfig';
import { createMockFileContent } from './createMockFile';
import { callBojagiStories } from './callBojagiStories';
import { createEntry } from './createEntry';
import { getMocksPath } from './pathFactories';

import { createExportFnFactory } from './createExportFn';

const writeFilePromise = promisify(fs.writeFile);
const mkdirPromise = promisify<string, { recursive: boolean }, void>(fs.mkdir as any);
export const createExportFn = createExportFnFactory(addProps('Bojagi Collector'));

export default <CollectorMetadata>{
  name: 'Bojagi',
};

export const collector = async ({
  webpack,
  components,
  executionPath,
  storyFiles,
  projectWebpackConfig,
}: CollectorFunctionOptions) => {
  if (storyFiles.length === 0) {
    return;
  }
  const entry: Record<string, string> = createEntry(storyFiles);
  const componentPaths = components.map(component => component.filePath);
  const mockFileContent = createMockFileContent(getComponents());
  await mkdirPromise(getMocksPath(executionPath), { recursive: true });
  await Promise.all(
    mockFileContent.map(([path, content]) =>
      writeFilePromise(getMocksPath(executionPath, path.replace(/\//g, '__')), content)
    )
  );

  const config = getWebpackConfig({
    executionPath,
    entry,
    componentPaths,
    webpack,
    projectWebpackConfig,
  });

  const compiler = webpack(config as any);
  await runCompiler(compiler);

  callBojagiStories(executionPath, entry);
  writeRegisteredProps();
};

function runCompiler(compiler) {
  return new Promise((resolve, reject) =>
    compiler.run((err, compilerOutput) => {
      if (err || compilerOutput.compilation.errors.length > 0) {
        reject(err || compilerOutput.compilation.errors[0]);
      } else {
        resolve(compilerOutput);
      }
    })
  );
}
