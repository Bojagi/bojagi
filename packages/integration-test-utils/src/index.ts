import * as fs from 'fs';
import * as path from 'path';
import * as crypto from 'crypto';
import { execSync } from 'child_process';
import { getBojagiBin, getResultFolder } from './bojagiPaths';
import { snapShotFileJSON } from './snapshots';

export * from './snapshots';
export * from './preview';
export * from './bojagiPaths';

const TEST_RESULT_FOLDER = path.resolve(__dirname, 'tmp');

export function reset(basePath) {
  optionalRm(getResultFolder(basePath));
  optionalRm(TEST_RESULT_FOLDER);
  fs.mkdirSync(TEST_RESULT_FOLDER, { recursive: true });
}

export function withTestResult(name, fn) {
  return (...args) => {
    writeTestResult(name, args);
    return fn(...args);
  };
}

export function writeTestResult(name, content) {
  fs.writeFileSync(
    path.resolve(TEST_RESULT_FOLDER, `${name}.json`),
    JSON.stringify(content, null, '  ')
  );
}

export function getTestResult(name) {
  return JSON.parse(
    fs.readFileSync(path.resolve(TEST_RESULT_FOLDER, `${name}.json`), { encoding: 'uft8' })
  );
}

export function snapshotTestResult(name) {
  snapShotFileJSON(path.resolve(TEST_RESULT_FOLDER, `${name}.json`));
}

function optionalRm(filePath) {
  try {
    fs.rmdirSync(filePath, { recursive: true });
  } catch (e) {
    console.log(`${filePath} not found - skipping deletion`);
  }
}

export function hash(aString) {
  const h = crypto.createHash('sha256');
  h.update(aString);
  return h.digest('hex');
}

export function execBojagi(basePath, cmd) {
  return execSync(`${getBojagiBin(basePath)} ${cmd}`, {
    cwd: basePath,
  }).toString();
}
