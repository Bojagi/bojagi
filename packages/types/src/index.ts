export type Component = {
  filePath: string;
  gitPath: string;
  exportName: string;
  props: Record<string, any>[];
};

export type Module = {
  request: string;
  filePath?: string;
  gitPath?: string;
  isExternal: boolean;
  isNodeModule: boolean;
  packageName?: string;
  dependencies?: Module[];
};

export type ComponentExportDescription = {
  symbol: string;
  isDefaultExport: boolean;
};

export type EntrypointWithMetadata = {
  entrypoint: string;
  filePath: string;
  components: ComponentExportDescription[];
};

export type File = {
  name: string;
};

export type FileContent = {
  name: string;
  fileContent: string;
};

export type ComponentContent = {
  folder: string;
  fileContent: string;
};

export type ComponentWithMetadata = File & {
  fileContent: string;
  symbol: string;
  isDefaultExport: boolean;
  filePath: string;
  exportName: string;
  gitPath: string;
};
