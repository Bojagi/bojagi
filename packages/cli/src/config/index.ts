import * as fs from 'fs';
import { defaultConfig } from './defaultConfig';
import getCiSettingsFactory from './getCiSettings';
import { normaliseConfig } from './normaliseConfig';

import { getGitSettings } from './getGitSettings';
import { loadConfigFile } from './loadConfigFile';
import { Config, BaseConfig } from './types';

export * from './types';

const { PWD } = process.env;
export const CONFIG_FILE_PRIO = [
  {
    path: `${PWD}/.bojagirc.js`,
    fn: loadJsConfig,
  },
  {
    path: `${PWD}/.bojagirc.json`,
    fn: loadJsonConfig,
  },
  {
    path: `${PWD}/.bojagirc`,
    fn: loadJsonConfig,
  },
];

const getCiSettings = getCiSettingsFactory(process.env);

export const getConfig: (customConfig: Partial<BaseConfig>) => Promise<Config> = async (
  customConfig = {}
) => {
  console.log('XXXXXXXXXX');

  const configFile = loadConfigFile({ configFilePrio: CONFIG_FILE_PRIO, fs });
  console.log('get me', defaultConfig);

  return normaliseConfig({
    ...(await getGitSettings(configFile.executionPath || defaultConfig.executionPath)),
    ...defaultConfig,
    ...configFile,
    ...getCiSettings(),
    ...customConfig,
  });
};

function loadJsConfig(path) {
  return require(path);
}

function loadJsonConfig(path) {
  return JSON.parse(fs.readFileSync(path).toString());
}
