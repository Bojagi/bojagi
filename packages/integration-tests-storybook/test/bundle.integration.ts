import { deleteTmpFolder, execBojagi, snapShotTmpFolder } from '@bojagi/integration-test-utils/src';
import { EXEC_PATH } from './config';

test('run bundle command', () => {
  deleteTmpFolder(EXEC_PATH);
  execBojagi(EXEC_PATH, 'bundle');
  snapShotTmpFolder(EXEC_PATH);
});
