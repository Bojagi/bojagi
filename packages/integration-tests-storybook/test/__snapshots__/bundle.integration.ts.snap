// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run bundle command 1`] = `
"{
 \\"version\\": \\"3\\",
 \\"namespaces\\": [
  {
   \\"name\\": \\"default\\",
   \\"framework\\": {
    \\"name\\": \\"react\\",
    \\"version\\": \\"16.12.0\\"
   }
  }
 ]
}"
`;

exports[`run bundle command 2`] = `
"[
 {
  \\"name\\": \\"vendors.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/vendors.js\\"
 },
 {
  \\"name\\": \\"Button.stories.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Button.stories.js\\"
 },
 {
  \\"name\\": \\"Header.stories.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Header.stories.js\\"
 },
 {
  \\"name\\": \\"Page.stories.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Page.stories.js\\"
 },
 {
  \\"name\\": \\"runtime~Button.stories.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/runtime~Button.stories.js\\"
 },
 {
  \\"name\\": \\"runtime~Header.stories.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/runtime~Header.stories.js\\"
 },
 {
  \\"name\\": \\"runtime~Page.stories.js\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/runtime~Page.stories.js\\"
 },
 {
  \\"name\\": \\"vendors.js.LICENSE.txt\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/vendors.js.LICENSE.txt\\"
 },
 {
  \\"name\\": \\"vendors.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/vendors.js.map\\"
 },
 {
  \\"name\\": \\"Button.stories.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Button.stories.js.map\\"
 },
 {
  \\"name\\": \\"Header.stories.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Header.stories.js.map\\"
 },
 {
  \\"name\\": \\"Page.stories.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/Page.stories.js.map\\"
 },
 {
  \\"name\\": \\"runtime~Button.stories.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/runtime~Button.stories.js.map\\"
 },
 {
  \\"name\\": \\"runtime~Header.stories.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/runtime~Header.stories.js.map\\"
 },
 {
  \\"name\\": \\"runtime~Page.stories.js.map\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/runtime~Page.stories.js.map\\"
 },
 {
  \\"name\\": \\"iframe.html\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/iframe.html\\"
 },
 {
  \\"name\\": \\"head.html\\",
  \\"namespace\\": \\"default\\",
  \\"outputFilePath\\": \\"files/head.html\\"
 }
]"
`;

exports[`run bundle command 3`] = `
"[
 {
  \\"title\\": \\"Example/Button\\",
  \\"fileName\\": \\"Button.stories\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"Primary\\",
    \\"storyName\\": \\"Primary\\"
   },
   {
    \\"exportName\\": \\"Secondary\\",
    \\"storyName\\": \\"Secondary\\"
   },
   {
    \\"exportName\\": \\"Large\\",
    \\"storyName\\": \\"Large\\"
   },
   {
    \\"exportName\\": \\"Small\\",
    \\"storyName\\": \\"Small\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"isExternal\\": true,
    \\"isNodeModule\\": false,
    \\"packageName\\": \\"react\\",
    \\"request\\": \\"react\\",
    \\"isCircularImport\\": false
   },
   {
    \\"filePath\\": \\"src/Button.js\\",
    \\"gitPath\\": \\"packages/integration-tests-storybook/src/Button.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"request\\": \\"./Button\\",
    \\"isCircularImport\\": false,
    \\"dependencies\\": [
     {
      \\"isExternal\\": true,
      \\"isNodeModule\\": false,
      \\"packageName\\": \\"react\\",
      \\"request\\": \\"react\\",
      \\"isCircularImport\\": false
     },
     {
      \\"filePath\\": \\"src/button.css\\",
      \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"request\\": \\"./button.css\\",
      \\"isCircularImport\\": false,
      \\"dependencies\\": [
       {
        \\"filePath\\": \\"src/button.css\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./button.css\\",
        \\"isCircularImport\\": true
       }
      ]
     }
    ]
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests-storybook/src/Button.stories.js\\",
  \\"name\\": \\"Button.stories\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/Button.stories.js\\",
  \\"files\\": [
   \\"runtime~Button.stories.js\\",
   \\"runtime~Button.stories.js.map\\",
   \\"vendors.js\\",
   \\"vendors.js.map\\",
   \\"Button.stories.js\\",
   \\"Button.stories.js.map\\"
  ]
 },
 {
  \\"title\\": \\"Example/Header\\",
  \\"fileName\\": \\"Header.stories\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"LoggedIn\\",
    \\"storyName\\": \\"Logged In\\"
   },
   {
    \\"exportName\\": \\"LoggedOut\\",
    \\"storyName\\": \\"Logged Out\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"isExternal\\": true,
    \\"isNodeModule\\": false,
    \\"packageName\\": \\"react\\",
    \\"request\\": \\"react\\",
    \\"isCircularImport\\": false
   },
   {
    \\"filePath\\": \\"src/Header.js\\",
    \\"gitPath\\": \\"packages/integration-tests-storybook/src/Header.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"request\\": \\"./Header\\",
    \\"isCircularImport\\": false,
    \\"dependencies\\": [
     {
      \\"isExternal\\": true,
      \\"isNodeModule\\": false,
      \\"packageName\\": \\"react\\",
      \\"request\\": \\"react\\",
      \\"isCircularImport\\": false
     },
     {
      \\"filePath\\": \\"src/Button.js\\",
      \\"gitPath\\": \\"packages/integration-tests-storybook/src/Button.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"request\\": \\"./Button\\",
      \\"isCircularImport\\": false,
      \\"dependencies\\": [
       {
        \\"isExternal\\": true,
        \\"isNodeModule\\": false,
        \\"packageName\\": \\"react\\",
        \\"request\\": \\"react\\",
        \\"isCircularImport\\": false
       },
       {
        \\"filePath\\": \\"src/button.css\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./button.css\\",
        \\"isCircularImport\\": false,
        \\"dependencies\\": [
         {
          \\"filePath\\": \\"src/button.css\\",
          \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"request\\": \\"./button.css\\",
          \\"isCircularImport\\": true
         }
        ]
       }
      ]
     },
     {
      \\"filePath\\": \\"src/header.css\\",
      \\"gitPath\\": \\"packages/integration-tests-storybook/src/header.css\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"request\\": \\"./header.css\\",
      \\"isCircularImport\\": false,
      \\"dependencies\\": [
       {
        \\"filePath\\": \\"src/header.css\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/header.css\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./header.css\\",
        \\"isCircularImport\\": true
       }
      ]
     }
    ]
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests-storybook/src/Header.stories.js\\",
  \\"name\\": \\"Header.stories\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/Header.stories.js\\",
  \\"files\\": [
   \\"runtime~Header.stories.js\\",
   \\"runtime~Header.stories.js.map\\",
   \\"vendors.js\\",
   \\"vendors.js.map\\",
   \\"Header.stories.js\\",
   \\"Header.stories.js.map\\"
  ]
 },
 {
  \\"title\\": \\"Example/Page\\",
  \\"fileName\\": \\"Page.stories\\",
  \\"storyItems\\": [
   {
    \\"exportName\\": \\"LoggedIn\\",
    \\"storyName\\": \\"Logged In\\"
   },
   {
    \\"exportName\\": \\"LoggedOut\\",
    \\"storyName\\": \\"Logged Out\\"
   }
  ],
  \\"dependencies\\": [
   {
    \\"isExternal\\": true,
    \\"isNodeModule\\": false,
    \\"packageName\\": \\"react\\",
    \\"request\\": \\"react\\",
    \\"isCircularImport\\": false
   },
   {
    \\"filePath\\": \\"src/Page.js\\",
    \\"gitPath\\": \\"packages/integration-tests-storybook/src/Page.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"request\\": \\"./Page\\",
    \\"isCircularImport\\": false,
    \\"dependencies\\": [
     {
      \\"isExternal\\": true,
      \\"isNodeModule\\": false,
      \\"packageName\\": \\"react\\",
      \\"request\\": \\"react\\",
      \\"isCircularImport\\": false
     },
     {
      \\"filePath\\": \\"src/Header.js\\",
      \\"gitPath\\": \\"packages/integration-tests-storybook/src/Header.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"request\\": \\"./Header\\",
      \\"isCircularImport\\": false,
      \\"dependencies\\": [
       {
        \\"isExternal\\": true,
        \\"isNodeModule\\": false,
        \\"packageName\\": \\"react\\",
        \\"request\\": \\"react\\",
        \\"isCircularImport\\": false
       },
       {
        \\"filePath\\": \\"src/Button.js\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/Button.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./Button\\",
        \\"isCircularImport\\": false,
        \\"dependencies\\": [
         {
          \\"isExternal\\": true,
          \\"isNodeModule\\": false,
          \\"packageName\\": \\"react\\",
          \\"request\\": \\"react\\",
          \\"isCircularImport\\": false
         },
         {
          \\"filePath\\": \\"src/button.css\\",
          \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"request\\": \\"./button.css\\",
          \\"isCircularImport\\": false,
          \\"dependencies\\": [
           {
            \\"filePath\\": \\"src/button.css\\",
            \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
            \\"isExternal\\": false,
            \\"isNodeModule\\": false,
            \\"request\\": \\"./button.css\\",
            \\"isCircularImport\\": true
           }
          ]
         }
        ]
       },
       {
        \\"filePath\\": \\"src/header.css\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/header.css\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./header.css\\",
        \\"isCircularImport\\": false,
        \\"dependencies\\": [
         {
          \\"filePath\\": \\"src/header.css\\",
          \\"gitPath\\": \\"packages/integration-tests-storybook/src/header.css\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"request\\": \\"./header.css\\",
          \\"isCircularImport\\": true
         }
        ]
       }
      ]
     },
     {
      \\"filePath\\": \\"src/page.css\\",
      \\"gitPath\\": \\"packages/integration-tests-storybook/src/page.css\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"request\\": \\"./page.css\\",
      \\"isCircularImport\\": false,
      \\"dependencies\\": [
       {
        \\"filePath\\": \\"src/page.css\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/page.css\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./page.css\\",
        \\"isCircularImport\\": true
       }
      ]
     }
    ]
   },
   {
    \\"filePath\\": \\"src/Header.stories.js\\",
    \\"gitPath\\": \\"packages/integration-tests-storybook/src/Header.stories.js\\",
    \\"isExternal\\": false,
    \\"isNodeModule\\": false,
    \\"request\\": \\"./Header.stories\\",
    \\"isCircularImport\\": false,
    \\"dependencies\\": [
     {
      \\"isExternal\\": true,
      \\"isNodeModule\\": false,
      \\"packageName\\": \\"react\\",
      \\"request\\": \\"react\\",
      \\"isCircularImport\\": false
     },
     {
      \\"filePath\\": \\"src/Header.js\\",
      \\"gitPath\\": \\"packages/integration-tests-storybook/src/Header.js\\",
      \\"isExternal\\": false,
      \\"isNodeModule\\": false,
      \\"request\\": \\"./Header\\",
      \\"isCircularImport\\": false,
      \\"dependencies\\": [
       {
        \\"isExternal\\": true,
        \\"isNodeModule\\": false,
        \\"packageName\\": \\"react\\",
        \\"request\\": \\"react\\",
        \\"isCircularImport\\": false
       },
       {
        \\"filePath\\": \\"src/Button.js\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/Button.js\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./Button\\",
        \\"isCircularImport\\": false,
        \\"dependencies\\": [
         {
          \\"isExternal\\": true,
          \\"isNodeModule\\": false,
          \\"packageName\\": \\"react\\",
          \\"request\\": \\"react\\",
          \\"isCircularImport\\": false
         },
         {
          \\"filePath\\": \\"src/button.css\\",
          \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"request\\": \\"./button.css\\",
          \\"isCircularImport\\": false,
          \\"dependencies\\": [
           {
            \\"filePath\\": \\"src/button.css\\",
            \\"gitPath\\": \\"packages/integration-tests-storybook/src/button.css\\",
            \\"isExternal\\": false,
            \\"isNodeModule\\": false,
            \\"request\\": \\"./button.css\\",
            \\"isCircularImport\\": true
           }
          ]
         }
        ]
       },
       {
        \\"filePath\\": \\"src/header.css\\",
        \\"gitPath\\": \\"packages/integration-tests-storybook/src/header.css\\",
        \\"isExternal\\": false,
        \\"isNodeModule\\": false,
        \\"request\\": \\"./header.css\\",
        \\"isCircularImport\\": false,
        \\"dependencies\\": [
         {
          \\"filePath\\": \\"src/header.css\\",
          \\"gitPath\\": \\"packages/integration-tests-storybook/src/header.css\\",
          \\"isExternal\\": false,
          \\"isNodeModule\\": false,
          \\"request\\": \\"./header.css\\",
          \\"isCircularImport\\": true
         }
        ]
       }
      ]
     }
    ]
   }
  ],
  \\"gitPath\\": \\"packages/integration-tests-storybook/src/Page.stories.js\\",
  \\"name\\": \\"Page.stories\\",
  \\"namespace\\": \\"default\\",
  \\"filePath\\": \\"src/Page.stories.js\\",
  \\"files\\": [
   \\"runtime~Page.stories.js\\",
   \\"runtime~Page.stories.js.map\\",
   \\"vendors.js\\",
   \\"vendors.js.map\\",
   \\"Page.stories.js\\",
   \\"Page.stories.js.map\\"
  ]
 }
]"
`;
